### ISSUE TITLE
Filter error

### ISSUE THREAD

**Issue by anderson-fachini** [iVantage/angular-ivh-multi-select#11]
I noticed that in a specific situation the filter function doesn't work well.
As it is not so simple to explain, I created a gif showing the problem and a page on plnkr that simulates the problem.

![ivh](https://cloud.githubusercontent.com/assets/7707758/14212953/b7615a10-f80b-11e5-835f-1f03f7cec0c6.gif)

https://embed.plnkr.co/chy1jYG6p0VjFQA131PM/

**Comment by jtrussell** [comment]
Funky, we'll take a look - thanks for the gif and demo!

**Comment by jtrussell** [comment]
I was really scratching my head over this one at first but it's actually quite simple.

This filter just uses Angular's [`filterFilter`](https://docs.angularjs.org/api/ng/filter/filter) which considers **all** fields of the objects in your collection. In particular it considers the boolean `selected` field...

So, initially your collection looks something like this:

```
[
  { filialNome: 'abc', selected: false },
  { filialNome: 'def', selected: false },
  { filialNome: 'ghi', selected: false }
]
```

When you enter `'f'` into the search, `filterFilter` matches each item because `false` gets internally [cast to a string](https://github.com/angular/angular.js/blob/master/src/ng/filter/filter.js#L188-L190) (i.e. `'false'`) by the filter.

After all the items are set set to being selected, the things which don't have an `'f'` in their label are hidden because `'true'.indexOf('f') === -1`.

Yuck.

We can fix this but it'll probably mean using our own filter and just considering the label field... we'll have to do something special for the cases where a label expression is provided.

**Comment by jtrussell** [comment]
Should be fixed in v0.8.0 :smile:
